### ISSUE TITLE
ComboBox bugs and missing light prop

### ISSUE THREAD

**Issue by nicoxxie** [IBM/carbon-components-react#1100]
## Detailed description

> Describe in detail the issue you're having.
1) Cancel button only shown when dropdown is expanded
![screen shot 2018-07-18 at 11 03 01 am](https://user-images.githubusercontent.com/17915105/42893536-344f3d8c-8a7a-11e8-810a-d8d26e342855.png)

2) If an option is already selected, the dropdown list only contains that one item
![screen shot 2018-07-18 at 11 03 06 am](https://user-images.githubusercontent.com/17915105/42893544-3c446bac-8a7a-11e8-9ed2-be7be87f9851.png)

![screen shot 2018-07-18 at 11 03 15 am](https://user-images.githubusercontent.com/17915105/42893553-4549ba7c-8a7a-11e8-9bb3-836f1f587e83.png)

3) missing 'light' option

> Is this a feature request (new component, new icon), a bug, or a general issue?
bug

> Is this issue related to a specific component?
ComboBox

> What did you expect to happen? What happened instead? What would you like to see changed?
1) Cancel button shown even if dropdown is not expanded
2) dropdown shows list of all options, with the current selected value highlighted.
3) light background

> What browser are you working in?
all

> What version of the Carbon Design System are you using?
  "carbon-components": "9.1.20",
    "carbon-components-react": "6.12.2",

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
ibm cloud

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

## Additional information

* Screenshots or code
* Notes

**Comment by lovemecomputer** [comment]
@IBM/carbon-designers the "bug" described here is the intentional UX, though perhaps it looks a little confusing when an item is selected? it appears the UX of this component is functioning according to spec.

**Comment by nicoxxie** [comment]
@lovemecomputer If the user wants to select a value, let's say the default value is 'No access'
The user has to: 
1. Click the dropdown (so the x button is shown)
2. Click the x (then dropdown shrunks)
3. Click the dropdown again
3. Select the new value

If the user makes a mistake, he/she has to repeat the step again.  
Not a ux pro here but this doesn't seem very user friendly.  You are right it's not a 'bug' *bug*, but should be *improvement* bug

**Comment by lovemecomputer** [comment]
@nicoxxie thanks for the detail! 

i think we need to get some input from @IBM/carbon-designers , perhaps we need to rethink the UX of this component, or better sync up on how it is implemented. the original issues you posted, 1 & 2 function according the design assets that we have.

i believe issue 4, the 'value/label' and 'id' are functioning properly, but the taxonomy is perhaps awkward because it refers to the conceptual idea of having an "id" and "label" for the information in the option. @joshblack i think you implemented this, and if something needs to change i'd like to sync to to make sure i understand the context.

for now, @aledavila and i put in this PR to add the light option in PR https://github.com/IBM/carbon-components-react/pull/1387

**Comment by lovemecomputer** [comment]
@IBM/carbon-designers @chrisconnors-ibm  @IBM/carbon-developers : it seems like this `ComboBox` is an example of a component that is complex enough that we need to capture more detail on the specification of the UX of interacting with it. the spec of the `ComboBox` is still images, and the react implementation matches it, but the component seems 

perhaps this can be shared in the specs on the component webpage too! @asudoh pointed out that there are good animated examples of capturing how a component functions at: https://material.io/design/components/dialogs.html#full-screen-dialog . something like a storyboard, with redlines and highlighted click targets, might also work.

i can schedule a meeting for next week to talk how we might want to share these kinds of details with each other!

**Comment by aagonzales** [comment]
Yes, I think we need to do some digging into this topic and maybe re-evaluate the interaction of the component as a whole. Unfortunately, the main designer of this component is no longer here so we can get quick validation on this. But I believe part of the reason it only shows the one is because this is a "filter" and not just a selection. 

Attaching the old design issues for future reference.
https://github.ibm.com/carbon/issues/issues/313
https://github.ibm.com/carbon/issues/issues/277

**Comment by nicoxxie** [comment]
@aagonzales We were using 'Typeahead' and was told to use 'ComboBox' as typeahead was deprecated.  Is there a more appropriate component we can use? that allows type and selecting?

**Comment by alexartavia11** [comment]
Hello. My team is working on this issue and we have a question moving forward: 

Regarding point 4 (Value should be id), Could you explain what's the point of having the input value set to the item id please? Remember the input is also used for the typeahead functionality and the input value is what's shown to the user. 

@IBM/carbon-developers @IBM/carbon-designers

Thanks in advance for the answer!

**Comment by nicoxxie** [comment]
@alexartavia11 Hi normally input values could be retrieved by that dom id if someone choose to, but I guess in this case typeahead is a bit different...It's fine for us right now to not use the id since we use state controller forms.

**Comment by alexartavia11** [comment]
Hello @nicoxxie, Thanks a lot for the answer. We will proceed accordingly.
